### ISSUE TITLE
com.thoughtworks.xstream.core.util is not exported in OSGI manifest

### ISSUE THREAD

**Issue by laeubi** [x-stream/xstream#347]
`com.thoughtworks.xstream.io.xml.PrettyPrintWriter` uses `com.thoughtworks.xstream.core.util.QuickWriter` in the method:

`PrettyPrintWriter#writeText` but this type is not exported in the OSGi Manifest so one can't use this type in client code.

I'm using the latest release from here: 
https://mvnrepository.com/artifact/com.thoughtworks.xstream/xstream

it seems it is explicitly removed from export here:

https://github.com/x-stream/xstream/blob/d03f6987b793fac71ab89a31d7aa633c366c5289/xstream/pom.xml#L423

but I wonder what is the purpose of that?

Instead of experting the package, it mit be possible to have some abstraction or  move QuickWriter to the com.thoughtworks.xstream.io.xml package...

**Comment by joehni** [comment]
Yes, that package is for private use and does not provide any API stability. So, what's your use case for QuickWriter apart from the fact that it is in a protected method?

**Comment by laeubi** [comment]
If it is private I think it should only be used in private methods as otherwise it can drip into user-code anyways.
I'm currently using it to add a CDATA section into the emitted XML code like this:

```
CompactWriter compactWriter = new CompactWriter(writer) {
    boolean cdata = false;

    @Override
    public void startNode(String name, @SuppressWarnings("rawtypes") Class clazz) {
        super.startNode(name, clazz);
        cdata = name.equals("sectionText");
    }

    @Override
    protected void writeText(QuickWriter writer, String text) {
        if (cdata) {
            writer.write("<![CDATA[");
            writer.write(text);
            writer.write("]]>");
        } else {
            super.writeText(writer, text);
        }
    }
};
```

**Comment by joehni** [comment]
Since I cannot simply remove the methods for backward compatibility reasons, I have now deprecated them and added two new methods that can be overwritten instead:
```Java
String processAttributeValue(String);
String processText(String);
```

**Comment by joehni** [comment]
I revised this implementation, since it might have an impact on the performance writing the data and on the ability to write the data in chunks into the output stream even if not the complete text has been processed.

I took an alternate approach now and introduced the public interface RawWriter for the methods of a QuickWriter and overloaded the writeAttributeValue and writeText methods with alternatives using the new interface instead.
